src/app/products/manage-tax-components/create-tax-component/create-tax-component.component.html

```diff
@@ -0,0 +1,98 @@
+<div class="container">
+
+  <mat-card>
+
+    <form [formGroup]="taxComponentForm" (ngSubmit)="submit()">
+
+      <mat-card-content>
+
+        <div fxLayout="column">
+
+          <mat-form-field>
+            <mat-label>Name</mat-label>
+            <input matInput required formControlName="name">
+            <mat-error *ngIf="taxComponentForm.controls.name.hasError('required')">
+              Name is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Percentage</mat-label>
+            <input type="number" matInput required formControlName="percentage">
+            <mat-error *ngIf="taxComponentForm.controls.percentage.hasError('required')">
+              Percentage is <strong>required</strong>
+            </mat-error>
+            <mat-error *ngIf="taxComponentForm.controls.percentage.hasError('min')">
+              Percentage <strong>cannot</strong> be smaller than zero
+            </mat-error>
+            <mat-error *ngIf="taxComponentForm.controls.percentage.hasError('max')">
+              Percentage <strong>cannot</strong> be bigger than one hundred
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Debit Account Type</mat-label>
+            <mat-select formControlName="debitAccountType">
+              <mat-option *ngFor="let debitAccountType of debitAccountTypeData" [value]="debitAccountType.id">
+                {{ debitAccountType.value }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Debit Account</mat-label>
+            <mat-select formControlName="debitAcountId">
+              <mat-option *ngFor="let debitAccount of debitAccountData" [value]="debitAccount.id">
+                {{ debitAccount.name }}
+              </mat-option>
+            </mat-select>
+            <mat-error *ngIf="taxComponentForm.controls.debitAcountId.hasError('required')">
+              Debit account is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
```

Comment:
Similar to other components in the web app (for example create user), lets keep the control conditionally hidden and add/remove it accordingly from code instead of update. This way we can also ensure that `required` attribute is added to this field.

Do similar implementation for Credit Account.